api/rest/client.go

```diff
@@ -225,10 +234,9 @@ func (c *Client) sendWithRetries(ctx context.Context, req *http.Request, retryCo
 		c.rateLimiter.Update(ctx, response.StatusCode, response.Header)
 	}
 
-	if c.requestRetrier != nil && retryCount < c.requestRetrier.MaxRetries &&
-		c.requestRetrier.ShouldRetryFunc != nil && c.requestRetrier.ShouldRetryFunc(response) {
-		logger.V(1).Info(fmt.Sprintf("Retrying failed request %q (HTTP %s) after %d ms delay... (try %d/%d)", req.URL, response.Status, 100, retryCount+1, c.requestRetrier.MaxRetries), "statusCode", response.StatusCode, "try", retryCount+1, "maxRetries", c.requestRetrier.MaxRetries)
-		time.Sleep(100 * time.Millisecond)
+	if requestRetrier != nil && requestRetrier.ShouldRetry(response, retryCount) {
```

Comment:
```suggestion
	if  requestRetrier.ShouldRetry(response, retryCount) {
```
nit: 
cool, now you could also eliminate the `nil` check by checking for a `nil` receiver in `ShouldRetry()`

```
func (r *RequestRetrier) ShouldRetry(...) bool {
	if r == nil {
		return false
	}
```

Reply:
That might be a step too far.